src/Helpers/JWKFetcher.php

```diff
@@ -62,17 +77,43 @@ protected function convertCertToPem(string $cert) : string
         return $output;
     }
 
+    /**
+     * Get a specific kid from a JWKS.
+     *
+     * @param string      $kid     Key ID to get.
+     * @param string|null $jwksUri JWKS URI to use, or fallback on class-level one.
+     *
+     * @return mixed|null
+     */
+    public function getKey(string $kid, string $jwksUri = null)
+    {
+        $jwksUri = $jwksUri ?? $this->jwksUri;
+        $keys    = $this->getKeys( $jwksUri );
+
+        if (! empty( $keys ) && empty( $keys[$kid] )) {
+            $keys = $this->getKeys( $jwksUri, false );
+        }
+
+        return $keys[$kid] ?? null;
+    }
+
     /**
      * Gets an array of keys from the JWKS as kid => x5c.
      *
-     * @param string $jwks_url Full URL to the JWKS.
+     * @param string  $jwks_url  Full URL to the JWKS.
+     * @param boolean $use_cache Set to false to skip cache check; default true to use caching.
      *
      * @return array
      */
-    public function getKeys(string $jwks_url) : array
+    public function getKeys(string $jwks_url = null, bool $use_cache = true) : array
     {
+        $jwks_url = $jwks_url ?? $this->jwksUri;
+        if (empty( $jwks_url )) {
+            return [];
+        }
+
         $cache_key = md5($jwks_url);
-        $keys      = $this->cache->get($cache_key);
+        $keys      = $use_cache ? $this->cache->get($cache_key) : [];
         if (is_array($keys) && ! empty($keys)) {
```

Comment:
The only scenario in which this _could_ be valid is when `use_cache` is also true since you're hardcoding an empty array above, maybe it could be improved a bit. Also, why wouldn't the `cache->get` return an array?

Reply:
> maybe it could be improved a bit

I wasn't wild about it, TBH. I wanted to avoid getting the cache unless we're using it as well as nested ifs. I think the next commit should help. 

> Also, why wouldn't the cache->get return an array?

Cache primed incorrectly, malformed JWKS ...